### ISSUE TITLE
Removing the "All users enrolled" BBB role mapping does not prevent users from joining the BBB meeting

### ISSUE THREAD

**Issue by abias** [blindsidenetworks/moodle-mod_bigbluebuttonbn#244]
Steps to reproduce:

* Login as admin
* Create a course
* Create two users A and B
* Enrol user A into the course with the editing teacher role
* Enrol user B into the course with the student role

* Login as teacher
* Go to the course
* Add a BBB activity
* In the BBB activity instance settings, 
** Uncheck "Wait for moderator"
** Add a role mapping from the teacher role to the BBB moderator role
** Remove the role mapping from "All users enrolled" to the BBB viewer role
** For the result, see screenshot:
![image](https://user-images.githubusercontent.com/1092118/81291535-377b9300-906a-11ea-9b64-cfce0386f889.png)
* Create the activity finally

* Login as student
* Go to the course
* Go to the BBB activity

Expected result:
* As there is no explicit BBB role mapping for me, I am not allowed to join the meeting

Actual result:
* I can join the meeting without any problem as viewer

Proposal to resolve this discrepance:
* I think there is no reason to expect that the BBB role mapping would have access restriction effects
* But if the lack of a role mapping does not prevent you from joining a meeting, I would like to propose to add the "All users enrolled" -> "Viewers" mapping by default and make it unremovable.

**Comment by jfederico** [comment]
The issue here is that this Map is not (and it has never been) intended to be used as a way for authorizing users to join the meeting. It is enterely and exclusively for mapping Moodle User/Roles with BigBlueButton Roles.

Before this was implemented, statically and pretty much hardcoded, only Editing Teachers, Teachers and Administrators were Moderators, everybody else was Viewer. That is not costimizable at all.

So, we came up with the mapping idea, and later on with the mapping by default for the whole site through the Admin Settings. But this is how we think the mapping should look like

![image](https://user-images.githubusercontent.com/578359/87327761-7e6e8480-c502-11ea-92fd-4814d6c5aa6a.png)

And perhaps, in a next iteration we could even drop the whole concept in favour of the Moodle 'Assign roles' option
